src/PhpParser/Comparing/NodeComparator.php

```diff
@@ -34,6 +34,14 @@ public function printWithoutComments(Node | array | null $node): string
      */
     public function areNodesEqual(Node | array | null $firstNode, Node | array | null $secondNode): bool
     {
+        if ($firstNode instanceof Node && $secondNode === null) {
+            return false;
+        }
+
+        if ($secondNode instanceof Node && $firstNode === null) {
+            return false;
+        }
+
```

Comment:
It might be useful to add other comparisons too. E.g. one is array and other nul etc.

Reply:
Updated, I applied array vs null comparison and verify all are Nodes when it is array.